packages/react-router/src/useNavigate.tsx

```diff
@@ -21,22 +21,18 @@ export function useNavigate<
   TDefaultFrom extends string = string,
 >(_defaultOpts?: {
   from?: RoutePathsAutoComplete<RegisteredRouter['routeTree'], TDefaultFrom>
-}) {
-  const { navigate } = useRouter()
-
-  const matchPathname = useMatch({
-    strict: false,
-    select: (s) => s.pathname,
-  })
-
-  const result: UseNavigateResult<TDefaultFrom> = ({ from, ...rest }) => {
-    return navigate({
-      from: rest?.to ? matchPathname : undefined,
-      ...(rest as any),
-    })
-  }
+}): UseNavigateResult<TDefaultFrom> {
+  const { navigate, state } = useRouter()
 
-  return React.useCallback(result, [])
+  return React.useCallback(
+    (options: NavigateOptions) => {
+      return navigate({
+        ...options,
+        from: options?.to ? state.location.pathname : undefined,
```

Comment:
I am not too confident if `state.location.pathname` is correct here, or if we should use `state.location.resolvedPathname` instead? What's the difference ?

Reply:
also, question: could we rename `router.state` to `router.getState()` and make it a normal getter function? I was pretty confused to see that `router.state` is an imperative access. Looking at other state managers, there is often a `getState` method, like `store.getState()` in redux.

https://github.com/tanstack/router/blob/c9ef45761037f9807b7d6514b636a51f9079b9e4/packages/react-router/src/router.ts#L395-L397

To be non breaking, we would probably need to keep the `.state` accessor around

Reply:
> ... What's the difference ?

If I'm not mistaken `location` was for the intended destination and `resolvedLocation` was what the user was currently on.
I'd have to more thoroughly go through this (not on my phone) to confirm.

Reply:
> also, question: could we rename `router.state` to `router.getState()`...

There's honestly no real reason not to, but...

This would only be internal right? the public API for the existing hooks (useRouter and useRouterState) wouldn't see any changes right (like with redux, in user-land inside a component, I don't need to call a getter, just use the value returned from the hook)?

Maybe, we discuss this a bit more and add it in its own standalone release.
Depending on the final API, we can also add it onto the [v2 list](https://github.com/TanStack/router/discussions/975).

Reply:
Okay then I would switch to `resolvedLocation` here probably?

> This would only be internal right?

the router api is public via `useRouter`, so it would be breaking. But we can add the method and deprecate the getter and remove it in v2. Hooks wouldn't change.

Reply:
using `resolvedLocation` now: 03cfcc85

Reply:
Just for reference, the implementation for the `router.load` [method](https://github.com/TanStack/router/blob/876b887589b14fb4bce0773eb520417682a741e2/packages/react-router/src/router.ts#L1552), may be worth looking at since it's the only real place that it is functionally used (other than being a dependency for rendering).